### DISCUSSION TITLE
Influence on Microsoft Partner Center?

### DISCUSSION THREAD

**Discussion by baskren** [unoplatform/uno#18720] [General]
Team UNO!

First, you guys are awesome.  I'm starting to believe that Microsoft should give ownership of WinAppSDK to you guys because, in every step of the development / publication process, UNO is just more developer centric.  My guess is that you have a much better understanding of what it's like to be the consumer of a framework.

**Here's what I'm struggling with**: I've just spent a couple of weeks finding a non-mutually exclusive work around for [the WinAppSDK+UNO WebView2 issue](https://github.com/unoplatform/uno/issues/18552), [CSWinRT compile issue](https://github.com/unoplatform/uno/issues/18464), and using `SkiaSharp.Views.Windows.SKXamlCanvas` at the same time.  So imagine my disappointment when I submit the app to Partner Center and receive the following errors:

<img width="1092" alt="image" src="https://github.com/user-attachments/assets/8b87c10e-6e91-47cb-8951-75796727a120">

Well, it turns out that I'm sort of painted into a corner.  More recent versions of UNO use WindowsAppSDK 1.6 ... but it appears that brings more recent versions of `Microsoft.Windows.SDK.BuildTools` - [which Partner Center won't yet allow](https://github.com/microsoft/WindowsAppSDK/issues/4480#issuecomment-2379681380).  It looks like [you're already aware of this](https://github.com/microsoft/WindowsAppSDK/issues/4480#issuecomment-2267594558) so I thought a new discussion (rather than an new issue) might be appropriate.

Any insights or possible work arounds would be appreciated.

Thanks and 100% grateful for you and your team.

**Comment by baskren** [comment]
Hmmm.  For what it's worth,  I just tried adding the following to the `*.csproj` for each of the UNO projects in my app (there are 9 supporting UNO library projects and one UNO cross platform single project app project):

### ATTEMPT 1
```xml
 <WinAppSdkBuildToolsVersion>10.0.22621.0</WinAppSdkBuildToolsVersion>
  ```

**RESULT**: On all projects, I get the following errors:
```text
C:\Users\ben\Development\AWC\ConnectionCalc.WinUI\Bc3\P42.Uno.Controls\P42.Utils\P42.Utils.Uno\P42.Utils.Uno.csproj : warning NU1603: P42.Utils.Uno depends on Microsoft.Windows.SDK.BuildTools (>= 10.0.22621) but Microsoft.Windows.SDK.BuildTools 10.0.22621 was not found. An approximate best match of Microsoft.Windows.SDK.BuildTools 10.0.22621.1 was resolved.
C:\Users\ben\Development\AWC\ConnectionCalc.WinUI\Bc3\P42.Uno.Controls\P42.Utils\P42.Utils.Uno\P42.Utils.Uno.csproj : error NU1605: Warning As Error: Detected package downgrade: Microsoft.Windows.SDK.BuildTools from 10.0.22621.756 to 10.0.22621.1. Reference the package directly from the project to select a different version.
C:\Users\ben\Development\AWC\ConnectionCalc.WinUI\Bc3\P42.Uno.Controls\P42.Utils\P42.Utils.Uno\P42.Utils.Uno.csproj : error NU1605:  P42.Utils.Uno -> Microsoft.WindowsAppSDK 1.6.240829007 -> Microsoft.Windows.SDK.BuildTools (>= 10.0.22621.756)
C:\Users\ben\Development\AWC\ConnectionCalc.WinUI\Bc3\P42.Uno.Controls\P42.Utils\P42.Utils.Uno\P42.Utils.Uno.csproj : error NU1605:  P42.Utils.Uno -> Microsoft.Windows.SDK.BuildTools (>= 10.0.22621)
```

### ATTEMPT 2
```xml
 <WinAppSdkBuildToolsVersion>10.0.22621.756</WinAppSdkBuildToolsVersion>
  ```

**RESULT**: On the Single Project App Project, I get the following errors:
```text
"C:\Users\ben\Development\AWC\ConnectionCalc.WinUI\ConnectionCalc\ConnectionCalc.csproj" (Restore target) (1) ->
(Restore target) ->
  C:\Users\ben\Development\AWC\ConnectionCalc.WinUI\ConnectionCalc\ConnectionCalc.csproj : error NU1605: Warning As Error: Detected package downgrade: Microsoft.Windows.SDK.BuildTools from 10.0.26100.1 to 10.0.22621.756. Reference the package directly from the project to select a different version.
C:\Users\ben\Development\AWC\ConnectionCalc.WinUI\ConnectionCalc\ConnectionCalc.csproj : error NU1605:  ConnectionCalc -> Uno.Extensions.Core.WinUI 5.0.2 -> Microsoft.Windows.SDK.BuildTools (>= 10.0.26100.1)
C:\Users\ben\Development\AWC\ConnectionCalc.WinUI\ConnectionCalc\ConnectionCalc.csproj : error NU1605:  ConnectionCalc -> Microsoft.Windows.SDK.BuildTools (>= 10.0.22621.756)
```

To summarize, Using `<WinAppSdkBuildToolsVersion>` as a work around  seems to have worked in the UNO library projects but not on the Single Project App Project, because of a reference to `Uno.Extensions.Core.WinUI 5.0.2`?

**Comment by baskren** [comment]
OK, looks like I just got lucky!  I had `Toolkit` in my Single Project's `<UnoFeatures>` _and wasn't using it!_. I just deleted `Toolkit` from that list and I no longer am getting the NU1605 build error and nor the  `makepri.exe` version error during Partner Center package acceptance testing.

**Reply by jeromelaban** [reply]
Thanks for the update, indeed versions may be temporarily outdated, and so far we haven't found a list of available versions, but maybe we could add some validation using the wack tooling.
